src/lib/journey/callbacks/choice/choice.svelte

```diff
@@ -14,13 +15,30 @@
    * Details: Each callback is wrapped by the SDK to provide helper methods
    * for accessing values from the callbacks received from AM
    ************************************************************************* */
-  const choiceOptions = callback
-    .getChoices()
-    ?.map((label, idx) => ({ text: label, value: `${idx}` }));
+  const choiceOptions = callback.getChoices()?.map((text, idx) => ({
+    /**
+     * Since locale content keys for the choice component are built off of the
+     * values, there will not be any existing key-value pairs in the provided
+     * content. The third argument here, the original value, is what will be
+     * displayed. If you want to localize it, you'll need to add content keys
+     * in the locale file for that to override the original value.
+     */
+    text: interpolate(textToKey(text), null, text),
```

Comment:
Just want to make sure I understand this properly - we are providing the ability for a developer to define a set of choices in a choice node on the server, and localize this by providing matching values in a translation file within the app? Is that what's happening here?

Reply:
Yes, just to reiterate:

This framework prioritizes internal, localized content, when available. I'm doing this because having a single source of truth for your content is desirable. I want to minimize the amount of locale content that's spread between this and AM. Most companies have a separate team that manages the locale content, and they are used to dealing with the content within the projects themselves or locale (micro) services.

Now, there are some nodes in AM that neither provide a way to localize values NOR provide a desirable value to "key off of" for locale content. This is why I wrote the `textToKey` utility method. It provides a way to convert a regular string with spacing and punctuation into a normalized key. Platform UI does a same thing.

The choice node is one such example of a node that provides no way to localized the values and doesn't provide a natural key to use.

Now, the `interpolate` function works like this:

1. First argument is the key to look up the localized content string
2. Second is the data that is used for interpolation (injecting dynamic values into static strings)
3. Third is the fallback string, so if no content is found matching that key, it will use the text provided in this argument

There is also the chance when the key doesn't resolve to a localized string, and no fallback text is provided, it will convert the very key itself into a string. This is the fallback for the fallback :)